subprojects/packagefiles/zlib/meson.build

```diff
@@ -4,7 +4,7 @@ cc = meson.get_compiler('c')
 
 link_args = []
 compile_args = []
-if cc.get_id() == 'msvc'
+if cc.get_id() == 'msvc' or (cc.get_id().contains('clang') and host_machine.system() == 'windows')
```

Comment:
This still isn't right, there are some other compilers that will be hit here, really:
```meson
if cc.get_compiler_argument_syntax() == 'msvc' or (host_machine.system() == 'windows'  and cc.get_id() == 'clang')
```

Reply:
Could you please elaborate? I think even in the case this is actually mingw64 Clang, not the "official" Clang that is piggy-backing on MSVC headers defining -D_CRT_SECURE_NO_DEPRECATE and -D_CRT_NONSTDC_NO_DEPRECATE won't hurt, because those defines are not used in the zlib code, right?

Reply:
`cc.get_compiler_argument_syntax() == 'msvc'` catche all of the compilers that pretend to be msvc, which would be `msvc`, `clang-cl`, and `intel-cl`.

Reply:
sorry, I haven't noticed this change to `if cc.get_compiler_argument_syntax() == 'msvc'` in your comment on first sight and misinterpreted it; it's been a long day